Failing test on master:     --- FAIL: TestGenerateProgram/regress-11176

### What happened?

Looks like one of the test started failing, blocking PRs.

### Expected Behavior

Tests are green on master.

### Steps to reproduce

```
cd pkg/codegen/go/gen_program_test/batch5
go test -run TestGenerateProgram -test.v

--- FAIL: TestGenerateProgram (8.09s)
    --- PASS: TestGenerateProgram/multiline-string (1.16s)
    --- FAIL: TestGenerateProgram/regress-11176 (5.77s)
    --- PASS: TestGenerateProgram/throw-not-implemented (1.16s)
    --- SKIP: TestGenerateProgram/python-reserved (0.00s)
    --- SKIP: TestGenerateProgram/iterating-optional-range-expressions (0.00s)
    --- PASS: TestGenerateProgram/output-literals (0.00s)
    --- SKIP: TestGenerateProgram/dynamic-entries (0.00s)
    --- PASS: TestGenerateProgram/single-or-none (0.00s)

```

Logs give:

```
main imports
        github.com/pulumi/pulumi/sdk/v3/go/pulumi imports
        github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin imports
        google.golang.org/grpc/status imports
        google.golang.org/genproto/googleapis/rpc/status: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/status in multiple modules:
        google.golang.org/genproto v0.0.0-20230410155749-daa745c078e1 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto@v0.0.0-20230410155749-daa745c078e1/googleapis/rpc/status)
        google.golang.org/genproto/googleapis/rpc v0.0.0-20230706204954-ccb25ca9f130 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20230706204954-ccb25ca9f130/status)
main imports
        github.com/pulumi/pulumi/sdk/v3/go/pulumi imports
        github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin imports
        google.golang.org/grpc/codes tested by
        google.golang.org/grpc/codes.test imports
        google.golang.org/genproto/googleapis/rpc/code: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/code in multiple modules:
        google.golang.org/genproto v0.0.0-20230410155749-daa745c078e1 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto@v0.0.0-20230410155749-daa745c078e1/googleapis/rpc/code)
        google.golang.org/genproto/googleapis/rpc v0.0.0-20230706204954-ccb25ca9f130 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20230706204954-ccb25ca9f130/code)
main imports
        github.com/pulumi/pulumi/sdk/v3/go/pulumi imports
        github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin imports
        google.golang.org/grpc/status tested by
        google.golang.org/grpc/status.test imports
        google.golang.org/genproto/googleapis/rpc/errdetails: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/errdetails in multiple modules:
        google.golang.org/genproto v0.0.0-20230410155749-daa745c078e1 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto@v0.0.0-20230410155749-daa745c078e1/googleapis/rpc/errdetails)
        google.golang.org/genproto/googleapis/rpc v0.0.0-20230706204954-ccb25ca9f130 (/Users/anton.tayanovskyy/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20230706204954-ccb25ca9f130/errdetails)

```

### Output of `pulumi about`

$ go version                                                                                                                 
go version go1.20.1 darwin/arm64

### Additional context

Also failing in CI eg on the https://github.com/pulumi/pulumi/pull/13592 PR but I don't think it is related to the changes in the PR.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/codegen/testing/test/program_driver.go
+++ b/pkg/codegen/testing/test/program_driver.go
@@ -257,13 +257,11 @@ var PulumiPulumiProgramTests = []ProgramTest{
 		Directory:   "multiline-string",
 		Description: "Multiline string literals",
 	},
-	// TODO: TEMPORARILY DISABLED
-	// https://github.com/pulumi/pulumi/issues/13644
-	// {
-	// 	Directory:   "regress-11176",
-	// 	Description: "Regression test for https://github.com/pulumi/pulumi/issues/11176",
-	// 	Skip:        allProgLanguages.Except("go"),
-	// },
+	{
+		Directory:   "regress-11176",
+		Description: "Regression test for https://github.com/pulumi/pulumi/issues/11176",
+		Skip:        allProgLanguages.Except("go"),
+	},
 	{
 		Directory:   "throw-not-implemented",
 		Description: "Function notImplemented is compiled to a runtime error at call-site",

